.github/workflows/update.yaml

```diff
@@ -0,0 +1,112 @@
+name: Update Go Versions
+
+on:
+  schedule:
+    - cron: '0 22 * * 0-4' # 7:00 JST weekdays
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  update-go-versions:
+    runs-on: ubuntu-24.04
+    env:
+      GH_TOKEN: ${{ secrets.CYBOZU_NECO_PAT }}
+      UBUNTU_VERSIONS: "22.04 24.04"
+      GO_MAJOR_VERSIONS: "1.22 1.23"
+    
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Check for Go version updates
+        id: check-updates
+        run: |
+          NEED_UPDATE=0
+          touch ./update-body.txt
+          
+          # Get latest stable Go versions
+          GO_STABLE_VERSIONS=$(curl -sSLf https://go.dev/dl/?mode=json | jq -r '[.[] | select(.stable == true) | .version | sub("^go"; "")] | unique | .[]')
+          if [ -z "${GO_STABLE_VERSIONS}" ]; then
+            echo "Failed to fetch stable Go versions"
+            exit 1
+          fi
+
+          # Extract major and minor versions for comparison
+          GO_STABLE_MAJOR_VERSIONS=$(echo "${GO_STABLE_VERSIONS}" | cut -d. -f1,2 | sort -u)
+          
+          # Check for differences between GO_MAJOR_VERSIONS and GO_STABLE_MAJOR_VERSIONS
+          DIFF=$(diff <(printf "%s\n" ${GO_MAJOR_VERSIONS} | sort -V) <(printf "%s\n" ${GO_STABLE_MAJOR_VERSIONS} | sort -V))
+          if [ -n "${DIFF}" ]; then
+            echo "⚠️ **Major Version Update Required**" >> ./update-body.txt
+            echo "Differences detected between specified and stable Go versions:" >> ./update-body.txt
+            echo "${DIFF}" >> ./update-body.txt
+            echo "" >> ./update-body.txt
+          fi
+          
+          # Function to update Dockerfile and TAG file
+          update_files() {
+            local go_major=$1
+            local codename=$2
+            local latest_version=$3
+            local dockerfile_path="golang-all/golang-${go_major}-${codename}/Dockerfile"
+            local tag_path="golang-all/golang-${go_major}-${codename}/TAG"
+            if [ ! -f "${dockerfile_path}" ] || [ ! -f "${tag_path}" ]; then
+              echo "Files not found for Go ${go_major} on Ubuntu ${codename}"
+              return
+            fi
+            local current_version=$(grep 'ARG GO_VERSION=' "${dockerfile_path}" | cut -d= -f2)
+            if [ "${current_version}" != "${latest_version}" ]; then
+              NEED_UPDATE=1
+              # Update Dockerfile
+              sed -i "s/ARG GO_VERSION=.*/ARG GO_VERSION=${latest_version}/" "${dockerfile_path}"
+              # Update TAG file
+              echo "${latest_version}_${codename}" > "${tag_path}"
```

Comment:
The format of the TAG looks like `X.Y.Z.N-jammy`, where `X.Y.Z` is the Go version and N starts at 1 and increments. Therefore, it seems better to set it like `X.Y.Z.1_jammy` when updating.
https://github.com/cybozu/neco-containers/blob/8ce1748be91631f9cf478c22b17c06748abc04c4/golang-all/golang-1.22-jammy/TAG#L1